test/lib/whenParser.test.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+/*jshint expr: true*/
+
+const whenParser = require( '../../lib/whenParser' );
+
+const expect = require( 'chai' ).expect;
+
+const sinon = require( 'sinon' );
+
+
+describe( 'lib/whenParser', function() {
+
+    describe( '.parse', function() {
+        it( 'normal operation: is, then', function() {
+            const config = {
+                condition: 'a',
+                options: {
+                    is: {
+                        '@type': 'string',
+                        regex: /^(?!\s*$).+/,
+                    },
+                    then: {
+                        required: null,
+                    },
+                },
+            };
+
+            let whenSchema = { };
+            whenSchema.regex = sinon.stub().returns( whenSchema );
+
+            const engine = {
+                string: sinon.stub().returns( whenSchema ),
+                required: sinon.stub().returns( whenSchema ),
+            };
+
+            const [ condition, options ] = whenParser.parse( config, engine );
+
+            expect( engine.string.calledOnce ).to.be.true;
```

Comment:
The test verification doesn't seem to be the greatest way to prove it's working, or returns what you would expect it to. I'm still trying to think up a better way to show that the shape of options is what is expected.